includes/adapter/RecipientFromWCOrder.php

```diff
@@ -0,0 +1,164 @@
+
+use MyParcelNL\Sdk\src\Helper\ValidateStreet;
+use MyParcelNL\Sdk\src\Model\Recipient;
+use WC_Order;
+use WCMYPA_Admin;
+use WCMYPA_Settings;
+use WPO\WC\MyParcel\Compatibility\Order as WCX_Order;
+
+class RecipientFromWCOrder extends Recipient
+{
+    public const BILLING  = 'billing';
+    public const SHIPPING = 'shipping';
+    private const MIN_STREET_ADDITIONAL_INFO_LENGTH = 10;
+
+    /**
+     * Parameter $type should always be one of two constants, either 'billing' or 'shipping'.
+     *
+     * @param  \WC_Order $order
+     * @param  string    $originCountry
+     * @param  string    $type
+     *
+     * @throws \Exception
+     */
+    public function __construct(WC_Order $order, string $originCountry, string $type)
+    {
+        $recipientDetails = $this->createAddress($order, $type);
+        parent::__construct($recipientDetails, $originCountry);
+    }
+
+    /**
+     * @param  \WC_Order $order
+     * @param  string    $type
+     *
+     * @return array
+     * @throws \JsonException
+     */
+    private function createAddress(WC_Order $order, string $type): array
+    {
+        return [
+                'cc'          => $order->{"get_{$type}_country"}(),
+                'city'        => $order->{"get_{$type}_city"}(),
+                'company'     => $order->{"get_{$type}_company"}(),
+                'postal_code' => $order->{"get_{$type}_postcode"}(),
+                'region'      => $order->{"get_{$type}_state"}(),
+                'person'      => $this->getPersonFromOrder($order, $type),
+                'email'       => $this->getEmailAddressFromOrder($order),
+                'phone'       => $this->getPhoneNumberFromOrder($order),
+            ] + $this->getAddressFromOrder($order, $type);
+    }
+
+    /**
+     * @param  \WC_Order $order
+     * @param  string    $type
+     *
+     * @return array
+     * @throws \JsonException
+     */
+    private function getAddressFromOrder(WC_Order $order, string $type): array
+    {
+        $street               = WCX_Order::get_meta($order, "_{$type}_street_name") ?: null;
+        $number               = WCX_Order::get_meta($order, "_{$type}_house_number") ?: null;
+        $numberSuffix         = WCX_Order::get_meta($order, "_{$type}_house_number_suffix") ?: null;
+        $streetAdditionalInfo = $order->{"get_{$type}_address_2"}();
+
+        $isUsingSplitAddressFields = ! empty($street) || ! empty($number) || ! empty($numberSuffix);
+
+        if ($isUsingSplitAddressFields) {
+            $fullStreet           = implode(' ', [$street, $number, $numberSuffix]);
+
+            return [
+                'full_street'            => $fullStreet,
+                'street_additional_info' => $streetAdditionalInfo,
+            ];
+        }
+
+        $separatedStreet = $this->separateStreet($order->{"get_{$type}_address_1"}());
+        $streetAdditionalInfoIsNumberSuffix = strlen($order->{"get_{$type}_address_2"}()) < self::MIN_STREET_ADDITIONAL_INFO_LENGTH;
```

Comment:
Dit zou `addressLine2IsNumberSuffix` moeten heten want dat is eigenlijk wat je aan het checken bent